paasta_tools/contrib/rightsizer_soaconfigs_update.py

```diff
@@ -169,6 +169,34 @@ def get_recommendation_from_result(result, keys_to_apply):
     return rec
 
 
+def get_cassandra_recommendation_from_result(result, keys_to_apply):
+    rec = {}
+    for key in keys_to_apply:
+        val = result.get(key)
+        if not val or val == NULL:
+            continue
+        if key == "cpus":
+            rec["cpus"] = float(val)
+        elif key == "cpu_burst_percent":
+            rec["cpu_burst_percent"] = float(val)
+        elif key == "mem":
+            rec["mem"] = val
+        elif key == "disk":
+            rec["disk"] = val
+        elif key == "replicas":
+            rec["replicas"] = int(val)
+    return rec
+
+
+def get_recommendation_from_result(instance_type, result, keys_to_apply):
```

Comment:
i would normally suggest that all new functions have type annotations - but the existing code is untyped (and uses largely the same types as the new function), so I won't do that here :p

that said, I don't think it'd be a particularly large refactor for future enterprising yelpers:
```
SupportedInstanceType = Literal["kubernetes", "eks"]

ConfigKey = Literal[...]
SUPPORTED_CSV_KEYS = typing.get_args(ConfigKey)

class CassandraRightsizerResult(TypedDict):
    ...
class KubernetesRightsizerResult(TypedDict):
    ...
class CassandraRecommendation(TypedDict):
    ...
class KubernetesRecommendation(TypedDict):
    ...

def get_recommendation_from_result(
    instance_type: SupportedInstanceType,
    result: CassandraRightsizerResult | KubernetesRightsizerResult,
    keys_to_apply: list[ConfigKey]
) -> CassandraRecommendation | KubernetesRecommendation:
    ...
```

and so on :)